.github/workflows/main_workflow.yml

```diff
@@ -1,19 +1,40 @@
-name: CI
+name: Java CI
 
-on: [pull_request]
+on: [push, pull_request]
 
 jobs:
   build:
-
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
+      - uses: actions/checkout@v2
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+
+      - name: Run java checkstyle
+        run: ./gradlew autoLintGradle -x test -x jacocoTestReport
+
+      - name: Set up JDK 11
```

Comment:
Why do you setup JDK after the Gradle started? :) How would Gradle work without JDK?